promsonnet/README.md

```diff
@@ -68,3 +73,39 @@ $ jsonnet example.jsonnet
 $ # or:
 $ jsonnet patch.jsonnet
 ```
+> NOTE: This `v1.patchRule` function is intelligent: it can work with PromSonnet created rule groups
+as well as ones that were created manually (without the `groups_map` for example).
+
+## Working with Mixins
+Given that we often want to patch existing rules, and those often exist within mixins,
+we can also do:
+```
+mixins+: prom.v1.mixin.patchRule('base', 'prometheus_metamon', 'PrometheusDown', { 'for': '10m' }),
+```
+This will apply the same patch to the `base` mixin.
+
+## 'Upgrading' existing rule groups and mixins
```

Comment:
The maps are useful, totally agree that it is the way forward, but I would see it as a layer on top of the current array model instead of trying to upgrade the array model to match the layer on top. This upgrade process follows a weird process of  `array -> map -> array` because eventually Prometheus consumes an array.

Maybe the promsonnet should have a `asLegacyMixin()` function that converts the internal maps to arrays for use with the current mixin model? That way the process works in one direction `maps -> array`.

Reply:
Or maybe the `update()` just needs to be called inside patchRule and other functions where relevant? Taking the burden and complexity away from the consumer.

If so, there is even more reason not to 'update' as the `array -> map -> array` approach is now part of the promsonnet code, making it even harder to reason about.

Reply:
there's no benefit to upgrading for the sake of `patchRule`, turns out it does not have a performance benefit. So the benefit to upgrade is just that a user might want to be able to patch rules manually, and this will allow them to do it. I don't expect it to be a highly used feature, but who knows.